### ISSUE TITLE
Sporadic failure to start looting

### ISSUE THREAD

**Issue by SteveTownsend** [SteveTownsend/SmartHarvestSE#307]
Nexus reports continue

> 07 July 2021, 9:19AM
> The new version doesn't work at all because it doesn't collect anything even though everything is activated
> Even pausing and restarting does not help
> I also reinstalled everything, including a new Game
> Even with a German Save, it doesn't work
> 
> I will probably have to throw Smart Harvest out completely und use the old working Version 3.7.0.15
> 
> I get only errors but the Mod doesn`t work:
> https://drive.google.com/file/d/1SIp10fHSMWxSOg-Zx0lWA29yfovrNgL2/view?usp=sharing

> 06 July 2021, 1:15PM
> I tried loading both full saves and SSE Engine Fixes converted quicksaves to regular but it still doesn't always work
> SmartHarvestSE.log full save load
> Anything else that can be done?

**Comment by SteveTownsend** [comment]
I had this happen in a cycle of quick load on near-vanilla. The VM check from the plugin hangs and is not reset, if script response was pending when game load kicked off. Logs of that:

	Line 553094: 2021-07-07 20:03:27.809    trace  17656 skip REFR 0x00086800, 3D not loaded /0x00000021
	Line 553095: 2021-07-07 20:03:27.809    trace  17656 omit out of range REFR 0x00066fd7()
	Line 553096: 2021-07-07 20:03:27.809    trace  17656 omit out of range REFR 0x0006715a()
	Line 553097: 2021-07-07 20:03:27.809     info  17656 Sort and truncate 0 eligible REFRs
	Line 553098: 2021-07-07 20:03:27.809     info  17656 wait for 1000 milliseconds
	Line 553100: 2021-07-07 20:03:28.809     info  17656 UI status request # 21

Thread 17656 never wakes up because the kick for the condition variable never arrives. Added logic to _UIState::Reset_ to release that pending call, if it's found to be waiting.

**Comment by SteveTownsend** [comment]
Fix seems to be working, 10+ reloads did not cause the hang vs a repro after 2 or 3 before.

**Comment by SteveTownsend** [comment]
This fix is good but still issues. Here, settings cache is not refreshed until after scan thread starts up, leaving a wait in progress with an uninitialized interval
```
2021-07-08 20:14:52.128     info  12112 Starting SHSE Worker Thread
2021-07-08 20:14:52.553     info   4200 SHSE_MCM.OnGameReload starting
2021-07-08 20:14:52.598     info   4200 Loaded C:\Program Files (x86)\Steam\steamapps\common\Skyrim Special Edition\Data\SKSE\Plugins\SmartHarvestSE.Defaults.ini OK
2021-07-08 20:14:57.171     info  10652 SHSE_MCM.OnGameReload starting
2021-07-08 20:14:57.171     info  10652 Migrate FormList size(0) to Form[]
2021-07-08 20:14:57.204     info   4200 Migrate FormList size(0) to Form[]
2021-07-08 20:15:04.160     info  11280 Reference Search unpaused
2021-07-08 20:15:09.329     info   7456 Reference Search unpaused
2021-07-08 20:15:09.397     info  11280 Reference Search disabled
2021-07-08 20:15:09.397    trace  11280 Refresh settings cache
```
Fixes applied for this in v3.9.4.1
